_drafts/making-backend-scalable.markdown

```diff
@@ -0,0 +1,130 @@
+---
+layout: post
+title: "Making Autolab's Backend Scalable"
+author: Ilter Canberk
+
+tags: [tango, redis, scalability]
+
+---
+
+[Tango](https://github.com/autolab/Tango) is a stand-alone, RESTful service that Autolab uses as the back-end for autograding. Tango receives grading jobs from Autolab's front-end, adds them in a job queue, assigns them to available containers for grading and shepherds the job throuout the process. In its early days, Tango was mostly used for jobs that ran in under 5 seconds. However, over the recent semesters, Autolab has grown to host classes like Distributed Systems, Machine Learning. Some of these classes have assessments that are computationally intensive on large datasets and contains bigger files.
+
+
```

Comment:
Considering that in the next paragraph you talk about the in-memory queue, all you'd need to do here is mention something like:

"As we looked into expanding our cluster of autograding machines, we were running into the problem of how to effectively schedule and manage queued jobs across our distributed backend. To this end, we're now using Redis to accomplish X, Y, and Z that our initial solution wasn't doing for us."